src/schema/position.rs

```diff
@@ -582,6 +582,26 @@ pub(crate) enum FieldDefinitionPosition {
     Union(UnionTypenameFieldDefinitionPosition),
 }
 
+impl PartialEq<CompositeTypeDefinitionPosition> for FieldDefinitionPosition {
```

Comment:
This should not be an eq implementation in my opinion. This means that `A.f == A`, which is not what I'd ever expect when reading an `==`. I think we can replace uses of this impl by `field_definition_position.parent() == composite_type_position` instead?

Reply:
Like the comment above, this was something that I thought I was going to need but didn't (I was trying to find my way through the maze of the codebase and got a bit lost). Sorry, I should have caught these. Unfortunately, the dead code lint doesn't catch unused trait impls (even for fully private types) :cry:

Reply:
great, no worries! 😄